### DISCUSSION TITLE
Win10 window refresh lag

### DISCUSSION THREAD

**Discussion by tscpp** [Seo-Rii/electron-acrylic-window#69] [Ideas]
https://github.com/sourcechord/FluentWPF/blob/b70731c6d026b2087dbf41656ff53854d48813d8/FluentWPF/Utility/AcrylicHelper.cs#L62

> In Windows10 1903 or later, if you use `ACCENT_ENABLE_ACRYLICBLURBEHIND`, it will not follow the mouse operation by dragging and moving the window.
> 
> Do not use `ACCENT_ENABLE_ACRYLICBLURBEHIND` until the behavior of the `SetWindowCompositionAttribute` function is fixed.

**Comment by seo-rii** [comment]
Yes, this is a bug in Windows. We introduced win10refresh.ts to minimize this bug.

**Comment by tscpp** [comment]
The win10refresh has some issues moving the window. Either the win10refresh should be fixed or temporary use the blur-behind filter.

**Comment by tscpp** [comment]
Maybe [WinUI 3 Preview 2](https://docs.microsoft.com/en-us/windows/apps/winui/winui3/) could be used instead of user32.dll?

**Comment by tscpp** [comment]
WinUI 3 is unfortunately not available as a c++ library for windows because it's made with XAML.

**Comment by tscpp** [comment]
As of WinUI 3 Project Reunion 0.5, it can be used with win32, directx and native c++: https://microsoft.github.io/microsoft-ui-xaml/about.html#controls

**Comment by seo-rii** [comment]
I'm trying to implement Mica Material. When the transparent option is given true, it works normally, but in other cases, it doesn't, so I'm looking for a solution.

**Reply by Void48** [reply]
Any luck on implementing Mica? :(

**Reply by seo-rii** [reply]
Sadly, no.

**Reply by Void48** [reply]
> Sadly, no.

@Seo-Rii hey I don't know if you saw but there's now public Win32 API for Mica,Acrylic effects
**DWMWA_SYSTEMBACKDROP_TYPE**

**Reply by Naozumi520** [reply]
Any news on this project? :(

**Reply by bakashigure** [reply]
Any news on this project? :((
